geode-management/src/main/java/org/apache/geode/management/configuration/GatewayReceiver.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.geode.cache.configuration.CacheElement;
+import org.apache.geode.cache.configuration.DeclarableType;
+import org.apache.geode.management.api.CorrespondWith;
+import org.apache.geode.management.api.RestfulEndpoint;
+import org.apache.geode.management.runtime.GatewayReceiverInfo;
+
+
+/**
+ * As a configuration object, user should not be able to set bind address and
```

Comment:
This class is external. So it needs a good class comment that describes to users how to use it. Those javadocs should point to the apis on ClusterManagementService that you would pass an instance of this class to.
It should also describe, in general, what the purpose of this class is.
I think this sentence it currently has should be removed. If you think it is needed in this context, explain at the end of the comment why setting the bind address and hostname on a gateway is not currently supported by this api.